#124028 - Fix stacked alerts, button width

ckeif-iat opened the pull request:
## Are you removing, renaming or moving a folder in this PR?
- [X] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [X] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

This addresses the remaining tasks on https://github.com/department-of-veterans-affairs/va.gov-team/issues/124028:

1. Do not stack alerts. Rather than showing an error alert followed by the email confirmation alert, now the email confirmation alert handles error state.

2. Update button size in alerts. Removed the `fullWidth` prop so the buttons are now all their default widths.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/124028

## Testing done

This impacts 3 areas in the same way:

#### Profile
- `/profile/contact-information`
  - `watch`: `yarn --cwd $( git rev-parse --show-toplevel ) watch --env entry=profile`
  - `mock`: `yarn mock-api --responses src/applications/personalization/profile/mocks/server.js`
  - mock files:
    - `PUT`: `src/applications/personalization/profile/mocks/server.js`

#### MHV landing Page
- `/my-health`
  - `watch`: `yarn --cwd $( git rev-parse --show-toplevel ) watch --env entry=mhv-landing-page`
  - `mock`: `yarn mock-api --responses src/applications/mhv-landing-page/mocks/api/index.js`
    - `PUT`: `src/applications/mhv-landing-page/mocks/api/index.js`

#### Dashboard
- `/my-va`
  - `watch`: `yarn --cwd $( git rev-parse --show-toplevel ) watch --env entry=dashboard`
  - `mock`:  `yarn mock-api --responses src/applications/personalization/dashboard/mocks/server.js`
    - `PUT`: `src/applications/personalization/profile/mocks/server.js`


## Screenshots

#### Profile

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |   <img width="504" height="708" alt="Screenshot 2025-11-18 at 11 20 06 AM" src="https://github.com/user-attachments/assets/17efd3c7-4576-48b1-bab4-1a4492ec4f19" />     |    <img width="502" height="745" alt="profile-mobile-chrome" src="https://github.com/user-attachments/assets/8e3a4e4a-6873-4942-89ec-b4b6317a7f5f" />   |
| Desktop |    <img width="1513" height="851" alt="Screenshot 2025-11-18 at 11 19 55 AM" src="https://github.com/user-attachments/assets/96a23f0e-d131-413f-9cd9-1f1ff59c86a0" />    |    <img width="1460" height="854" alt="profile-desktop-edge" src="https://github.com/user-attachments/assets/66948f54-662f-4af3-b5da-c56237f3c155" />   |

#### MHV Landing Page

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |    <img width="505" height="703" alt="Screenshot 2025-11-18 at 11 23 51 AM" src="https://github.com/user-attachments/assets/6cc5577d-d3fd-4a98-91bd-782f6393e4c4" />    |   <img width="408" height="718" alt="mhv-landing-mobile-edge" src="https://github.com/user-attachments/assets/57369e48-f1b6-4af2-8a7d-0002af2b89b8" />    |
| Desktop |    <img width="1512" height="853" alt="Screenshot 2025-11-18 at 11 23 56 AM" src="https://github.com/user-attachments/assets/1692f227-c7e3-409d-846a-d2196c602eee" />    |   <img width="1155" height="897" alt="mhv-landing-desktop-safari" src="https://github.com/user-attachments/assets/25909cba-0aaf-4b2e-93ed-29e5fd0ff5c4" />    |

#### Dashboard

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |    <img width="503" height="702" alt="Screenshot 2025-11-18 at 11 25 43 AM" src="https://github.com/user-attachments/assets/7ec765c2-df42-4e19-9268-95a3ad86dc93" />    |    <img width="406" height="711" alt="dashboard-mobile-firefox" src="https://github.com/user-attachments/assets/4bd72977-e534-490e-ad73-9e5be15f5dcf" />   |
| Desktop |     <img width="1512" height="852" alt="Screenshot 2025-11-18 at 11 25 32 AM" src="https://github.com/user-attachments/assets/8e0da58f-5bca-4fec-87ed-736d317ad6b6" />   |    <img width="1145" height="1028" alt="dashboard-desktop-chrome" src="https://github.com/user-attachments/assets/5da1b91f-bac5-4237-af70-7bb494b58d71" />   |

Additional Videos (with screenreader)

#### Chrome Desktop (profile - contact information)

https://github.com/user-attachments/assets/9f5d08ee-97ff-4835-b7a1-3ef06d808dde

#### Firefox mobile (MHV landing page)

https://github.com/user-attachments/assets/a9f9760a-8335-4b91-8a05-6a49cfd9799c

## What areas of the site does it impact?

This impacts the following paths:
* `profile/contact-information`
* `/my-va`
* `/my-health`

## Acceptance criteria

### Quality Assurance & Testing

- [X] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [X] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [X] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

ckeif-iat committed:
replace stacked alerts with single alert; fix button width

ckeif-iat committed:
refactor to use two separate components for the notify/error alert cases

ckeif-iat committed:
add tests for AlertConfirmAddContactEmailError

ckeif-iat committed:
add screenreader only span to new error components

ckeif-iat committed:
revert accidental text change

ckeif-iat committed:
Merge branch 'main' into 124028-remaining-email-confirmation-staging-review

carlosfelix2 commented on src/platform/mhv/tests/components/MhvAlertConfirmEmail.unit.spec.jsx:
```diff
@@ -225,8 +225,9 @@ describe('<MhvAlertConfirmEmail />', () => {
       await waitFor(() => {
-        getByTestId('mhv-alert--confirm-error');
-        getByTestId('mhv-alert--confirm-contact-email');
+        const alert = getByTestId('mhv-alert--confirm-error');
```
Consider using findByTestId and removing the wait.

carlosfelix2 commented on src/platform/mhv/tests/components/ProfileAlertConfirmEmailContent.spec.jsx:
```diff
@@ -0,0 +1,82 @@
+
+    await waitFor(() => {
+      clickButton(container, true);
+      expect(onEditClick.calledOnce).to.be.true;
```
You may need a wait here as well.

carlosfelix2 reviewed (changes requested):
Need to add e2e tests for this change

ckeif-iat committed:
Merge branch 'main' into 124028-remaining-email-confirmation-staging-review

ckeif-iat committed:
e2e tests and unit test enhancements

ckeif-iat commented on src/platform/mhv/tests/components/AlertConfirmAddContactEmailError.spec.jsx:
```diff
@@ -0,0 +1,68 @@
+
+    await waitFor(() => {
+      // Intro text
+      getByText(
```
Good call, cleaned these up to use `await findBy*(...)` instead of the `waitFor(...)`

carlosfelix2 commented on src/applications/mhv-landing-page/tests/e2e/confirm-email-address.cypress.spec.js:
```diff
@@ -0,0 +1,149 @@
+const MHV_EMAIL_CONFIRMATION_DISMISSED_COOKIE =
+  'MHV_EMAIL_CONFIRMATION_DISMISSED';
+
+const clickConfirmEmail = () => {
```
These functions should go in the page file `src/applications/mhv-landing-page/tests/e2e/pages/LandingPage.js`. This is pattern the VA wants. https://depo-platform-documentation.scrollhelp.site/developer-docs/best-practices-for-using-cypress#BestpracticesforusingCypress-PageObjects

carlosfelix2 commented on src/applications/mhv-landing-page/tests/e2e/confirm-email-address-no-email.cypress.spec.js:
```diff
@@ -0,0 +1,52 @@
+};
+
+const clickSkipAddingEmail = () => {
+  cy.get('va-alert[data-testid="mhv-alert--add-contact-email"]').within(() => {
```
Change this cy.get to cy.findByTestId

carlosfelix2 commented on src/applications/mhv-landing-page/tests/e2e/confirm-email-address-no-email.cypress.spec.js:
```diff
@@ -0,0 +1,52 @@
+  'MHV_EMAIL_CONFIRMATION_DISMISSED';
+
+const clickAddEmail = () => {
+  cy.get('va-alert[data-testid="mhv-alert--add-contact-email"]')
```
Change this cy.get to cy.findByTestId

carlosfelix2 commented on src/applications/mhv-landing-page/tests/e2e/confirm-email-address.cypress.spec.js:
```diff
@@ -0,0 +1,149 @@
+};
+
+const clickEditEmailLink = () => {
+  cy.get(
```
Can this be a cy.findByTestId?

carlosfelix2 commented on src/applications/mhv-landing-page/tests/e2e/confirm-email-address.cypress.spec.js:
```diff
@@ -0,0 +1,149 @@
+      emailAddress: 'user@test.com',
+    };
+    cy.login(userWithEmail);
+    cy.visit(rootUrl);
```
Same as the other file. Need a cy.wait, another cy.find, or change the gets above to finds.

carlosfelix2 commented on src/applications/mhv-landing-page/tests/e2e/confirm-email-address.cypress.spec.js:
```diff
@@ -0,0 +1,149 @@
+    clickConfirmEmail();
+    cy.wait('@updateEmail');
+
+    cy.get('[data-testid="mhv-alert--confirm-contact-email"]').should(
```
Change to findByTestId. Check everywhere else.

carlosfelix2 commented on src/applications/personalization/dashboard/tests/e2e/confirm-email-address.cypress.spec.js:
```diff
@@ -0,0 +1,165 @@
+    clickConfirmEmail();
+    cy.wait('@updateEmail');
+
+    cy.get('[data-testid="mhv-alert--confirm-contact-email"]').should(
```
This needs to be a find as the page may not be rendered yet even with the cy.wait above.

carlosfelix2 commented on src/applications/personalization/dashboard/tests/e2e/confirm-email-address.cypress.spec.js:
```diff
@@ -0,0 +1,165 @@
+    clickConfirmEmail();
+    cy.wait('@updateEmail');
+
+    cy.get('[data-testid="mhv-alert--confirm-contact-email"]').should(
```
This needs to be a find as the page may not be rendered yet even with the cy.wait above.

carlosfelix2 commented on src/applications/personalization/dashboard/tests/e2e/confirm-email-address.cypress.spec.js:
```diff
@@ -0,0 +1,165 @@
+    cy.wait('@updateEmail');
+
+    // Verify error alert is shown and cookie isn't set.
+    cy.get('[data-testid="mhv-alert--confirm-contact-email"]').should(
```
This needs to be a find as the page may not be rendered yet even with the cy.wait above.

carlosfelix2 commented on src/applications/personalization/dashboard/tests/e2e/confirm-email-address.cypress.spec.js:
```diff
@@ -0,0 +1,165 @@
+    cy.wait('@updateEmail');
+
+    // Verify success alert is shown and cookie is set.
+    cy.get('[data-testid="mhv-alert--confirm-contact-email"]').should(
```
This needs to be a find as the page may not be rendered yet even with the cy.wait above.

carlosfelix2 commented on src/applications/personalization/profile/tests/e2e/contact-information/confirm-email-address-no-email.cypress.spec.js:
```diff
@@ -0,0 +1,116 @@
+  cy.wait('@mockUser');
+};
+
+const clickAddEmailAddress = () => {
```
These functions should go in the page file `src/applications/mhv-landing-page/tests/e2e/pages/LandingPage.js`. This is pattern the VA wants. https://depo-platform-documentation.scrollhelp.site/developer-docs/best-practices-for-using-cypress#BestpracticesforusingCypress-PageObjects

carlosfelix2 commented on src/applications/personalization/profile/tests/e2e/contact-information/confirm-email-address-no-email.cypress.spec.js:
```diff
@@ -0,0 +1,116 @@
+  it('should focus the email address field when "Add a contact email" is clicked', () => {
+    clickAddEmailAddress();
+
+    cy.get('va-text-input[name="root_emailAddress"]')
```
Same as the other file. Need a cy.wait, another cy.find, or change the gets above to finds.

carlosfelix2 commented on src/applications/personalization/profile/tests/e2e/contact-information/confirm-email-address.cypress.spec.js:
```diff
@@ -0,0 +1,243 @@
+    cy.wait('@updateEmail');
+
+    // Confirm success alert shows after successful retry
+    cy.get('[data-testid="mhv-alert--confirm-contact-email"]').should(
```
This needs to be a find as the page may not be rendered yet even with the cy.wait above.

carlosfelix2 commented on src/applications/personalization/profile/tests/e2e/contact-information/confirm-email-address.cypress.spec.js:
```diff
@@ -0,0 +1,243 @@
+    cy.wait('@updateEmail');
+
+    // Confirm error alert appears in edit view
+    cy.get('[data-testid="mhv-alert--confirm-contact-email"]').should(
```
This needs to be a find as the page may not be rendered yet even with the cy.wait above.

carlosfelix2 commented on src/applications/personalization/profile/tests/e2e/contact-information/confirm-email-address.cypress.spec.js:
```diff
@@ -0,0 +1,243 @@
+
+    // Click the edit button and verify the email address field is focused
+    clickEditEmail();
+    cy.get('va-text-input[name="root_emailAddress"]')
```
Same as the other file. Need a cy.wait, another cy.find, or change the gets above to finds.

carlosfelix2 commented on src/applications/personalization/profile/tests/e2e/contact-information/confirm-email-address.cypress.spec.js:
```diff
@@ -0,0 +1,243 @@
+    clickConfirmEmail();
+    cy.wait('@updateEmail');
+
+    cy.get('[data-testid="mhv-alert--confirm-contact-email"]').should(
```
This needs to be a find as the page may not be rendered yet even with the cy.wait above.

carlosfelix2 reviewed (changes requested):
e2e changes requested

ckeif-iat committed:
code review feedback: clean up e2e tests

ckeif-iat commented:
Pushed [a change](https://github.com/department-of-veterans-affairs/vets-website/pull/40173/commits/3bc06374cfd95d15da7b1faf86bad740f652881a) addressing the e2e test feedback:

- Moved click functions into page objects
  - I couldn't find any pages for the dashboard and contact information tests, so I added some simple ones:
    - [src/applications/personalization/profile/tests/e2e/contact-information/pages/ContactInformationPage.js](https://github.com/department-of-veterans-affairs/vets-website/blob/3bc06374cfd95d15da7b1faf86bad740f652881a/src/applications/personalization/profile/tests/e2e/contact-information/pages/ContactInformationPage.js)
    - [src/applications/personalization/dashboard/tests/e2e/pages/Dashboard.js](https://github.com/department-of-veterans-affairs/vets-website/blob/3bc06374cfd95d15da7b1faf86bad740f652881a/src/applications/personalization/dashboard/tests/e2e/pages/Dashboard.js)
- Changed all the `.get(...data-testid=...)` calls to use `findByTestId(...)`. I think this should cover the need for `cy.wait(...)`
  - I kept the cy.get('va-text-input[name="root_emailAddress"]') calls since that's the best selector I could come up with. I think that should be fine though given the `findByTestId(...)` calls earlier in the test

ckeif-iat committed:
Merge branch 'main' into 124028-remaining-email-confirmation-staging-review

ckeif-iat committed:
Merge branch 'main' into 124028-remaining-email-confirmation-staging-review
